src/openforms/submissions/models/submission.py

```diff
@@ -412,7 +414,7 @@ def is_authenticated(self):
         return hasattr(self, "auth_info")
 
     @property
-    def registrator(self):
+    def registrator(self) -> "AuthInfo | RegistratorInfo | None":
```

Comment:
or `-> BaseAuthInfo | None`, but slightly incorrect as it either one or the other, never the base common class.

Reply:
can you use `from __future__ import annotations` so you can do this instead:

```suggestion
    def registrator(self) -> AuthInfo | RegistratorInfo | None:
```